File: smartadapter/src/test/java/smartadapter/internal/ViewHolderConstructorMapperTest.java
package smartadapter.internal;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

/*
 * Created by Manne Öhlund on 2019-07-19.
 * Copyright (c) All rights reserved.
 */

public class ViewHolderConstructorMapperTest {

    @Test
    public void testViewHolderConstructorMapper() {
        // Given
        List list = Arrays.asList(
                ReflectionUtilsTest.InnerClassViewHolder.class,
                ReflectionUtilsTest.StaticInnerClassViewHolder.class);
        ViewHolderConstructorMapper viewHolderConstructorMapper = new ViewHolderConstructorMapper();

        // When
        viewHolderConstructorMapper.add(list);
        viewHolderConstructorMapper.add(ReflectionUtilsTest.InnerClassViewHolder.class);

        // Then
        assertNotNull(viewHolderConstructorMapper.getConstructor(ReflectionUtilsTest.InnerClassViewHolder.class));
        assertNotNull(viewHolderConstructorMapper.getConstructor(ReflectionUtilsTest.StaticInnerClassViewHolder.class));
        assertNull(viewHolderConstructorMapper.getConstructor(ReflectionUtilsTest.InvalidInnerClassTestViewHolder.class));
    }
}

File: smartadapter/src/test/java/smartadapter/internal/ViewHolderConstructorMapperTest.kt
package smartadapter.internal

import org.junit.Assert.assertNotNull
import org.junit.Assert.assertNull
import org.junit.Test
import smartadapter.internal.mapper.ViewHolderConstructorMapper
import smartadapter.viewholder.SmartViewHolder
import kotlin.reflect.KClass

/*
 * Created by Manne Öhlund on 2019-07-19.
 * Copyright (c) All rights reserved.
 */

class ViewHolderConstructorMapperTest {

    @Test
    fun testViewHolderConstructorMapper() {
        // Given
        val list: MutableList<KClass<out SmartViewHolder<Any>>> = mutableListOf(
                ReflectionUtilsTest.InnerClassViewHolder::class,
                ReflectionUtilsTest.StaticInnerClassViewHolder::class)
        val viewHolderConstructorMapper = ViewHolderConstructorMapper()

        // When
        viewHolderConstructorMapper.add(list)
        viewHolderConstructorMapper.add(ReflectionUtilsTest.InnerClassViewHolder::class)

        // Then
        assertNotNull(viewHolderConstructorMapper.getConstructor(ReflectionUtilsTest.InnerClassViewHolder::class))
        assertNotNull(viewHolderConstructorMapper.getConstructor(ReflectionUtilsTest.StaticInnerClassViewHolder::class))
        assertNull(viewHolderConstructorMapper.getConstructor(ReflectionUtilsTest.InvalidInnerClassTestViewHolder::class))
    }
}